File: Queue/python/queue_.py
from node import Node

class Queue:
    def __init__(self, head = None, last =None):
        self.head = head
        self.last = last
        

    def is_empty(self):
        return self.head == None


    def enqueue(self, value):
        new_node = Node(value)
        if self.is_empty():
            self.head = new_node
            self.last = new_node
        else:
            self.last.next = new_node
            self.last = new_node


    def dequeue(self):
        if self.is_empty():
            raise IndexError('pop from empty queue')
        temp = self.head
        self.head = self.head.next
        return temp

File: Queue/c/queue.c
#include<stdio.h>
#include "queue.h"
#include<stdlib.h>

queue * initializerQueue(){
    queue * q = (queue*) malloc(sizeof(queue));
    q->init = 0;
    q->end = -1;
}


void enqueue(queue * q, int value){
    if(q->end == MAX - 1){
        printf("queue is full");
    }
    q->end ++;
    q->arr[q->end] = value;
}

int dequeue(queue* q){
    if(q->end == -1){
        printf("Queue is empty");
        return 0;
    }
    int temp = q->arr[q->init];
    for(int i = 0; i < q->end - 1; i++){
        q->arr[i] = q->arr[i + 1];
    }
    return temp;
}